unity-client/Assets/Scripts/MainScripts/DCL/Controllers/BuilderMode/BuilderInWorldController.cs

```diff
@@ -320,46 +403,68 @@ bool IsInsideTheLimits(SceneObject sceneObject)
 
         if (limits.bodies < usage.bodies + sceneObject.metrics.bodies)
         {
-            HUDController.i.buildModeHud.ShowSceneLimitsPassed();
+            HUDController.i.builderInWorldMainHud?.ShowSceneLimitsPassed();
             return false;
         }
 
         if (limits.entities < usage.entities + sceneObject.metrics.entities)
         {
-            HUDController.i.buildModeHud.ShowSceneLimitsPassed();
+            HUDController.i.builderInWorldMainHud?.ShowSceneLimitsPassed();
             return false;
         }
 
         if (limits.materials < usage.materials + sceneObject.metrics.materials)
         {
-            HUDController.i.buildModeHud.ShowSceneLimitsPassed();
+            HUDController.i.builderInWorldMainHud?.ShowSceneLimitsPassed();
             return false;
         }
 
         if (limits.meshes < usage.meshes + sceneObject.metrics.meshes)
         {
-            HUDController.i.buildModeHud.ShowSceneLimitsPassed();
+            HUDController.i.builderInWorldMainHud?.ShowSceneLimitsPassed();
             return false;
         }
 
         if (limits.textures < usage.textures + sceneObject.metrics.textures)
         {
-            HUDController.i.buildModeHud.ShowSceneLimitsPassed();
+            HUDController.i.builderInWorldMainHud?.ShowSceneLimitsPassed();
             return false;
         }
 
         if (limits.triangles < usage.triangles + sceneObject.metrics.triangles)
         {
-            HUDController.i.buildModeHud.ShowSceneLimitsPassed();
+            HUDController.i.builderInWorldMainHud?.ShowSceneLimitsPassed();
             return false;
         }
 
         return true;
     }
 
-    void CreateSceneObjectSelected(SceneObject sceneObject)
+    void OnSceneObjectSelected(SceneObject sceneObject)
     {
-        if (!IsInsideTheLimits(sceneObject)) return;
+        if (IsSceneObjectFloor(sceneObject))
```

Comment:
```suggestion
        if (IsSceneObjectFloor(sceneObject))
```

I think we can invert this if to reduce nesting.

Reply:
I think that this if has an else statement, I am missing something?